src/build_log.h

```diff
@@ -76,8 +86,14 @@ struct BuildLog {
              int start_time, int end_time, TimeStamp mtime);
   };
 
+#if __cplusplus >= 201103L || (defined(_MSC_VER) && _MSC_VER >= 1900)
+  typedef std::shared_ptr<LogEntry> LogEntryPtr;
```

Comment:
MIght as well use `using` if C++11 is available. It's superior to typedef in everyway.